api-specs/external-match-openapi.yaml

```
@@ -0,0 +1,354 @@
+      type: object
+      properties:
+        quote_mint:
+          type: string
+          description: Quote token address
+        base_mint:
+          type: string
+          description: Base token address
+        quote_amount:
+          type: integer
+          description: Amount of quote tokens
+        base_amount:
+          type: integer
+          description: Amount of base tokens
+        direction:
+          $ref: '#/components/schemas/OrderSide'
+
+    FeeTake:
+      type: object
+      properties:
+        relayer_fee:
+          type: integer
+          description: Fee paid to the relayer
+        protocol_fee:
+          type: integer
+          description: Fee paid to the protocol
+
+    ExternalOrder:
+      type: object
+      oneOf:
+        - required: [base_amount]
+          properties:
+            base_amount:
+              type: integer
+              description: Amount of base tokens
+              minimum: 1
+        - required: [quote_amount]
+          properties:
+            quote_amount:
+              type: integer
+              description: Amount of quote tokens
+              minimum: 1
+        - required: [exact_base_output]
+          properties:
+            exact_base_output:
+              type: integer
+              description: Exact base token output amount
+              minimum: 1
+        - required: [exact_quote_output]
+          properties:
+            exact_quote_output:
+              type: integer
+              description: Exact quote token output amount
+              minimum: 1
+      properties:
+        quote_mint:
+          type: string
+          description: Quote token address
+        base_mint:
+          type: string
+          description: Base token address
+        side:
+          $ref: '#/components/schemas/OrderSide'
+        min_fill_size:
+          type: integer
+          description: Minimum fill size
+          default: 0
+      required:
+        - quote_mint
+        - base_mint
+        - side
+
+    ApiExternalQuote:
+      type: object
+      properties:
+        order:
+          $ref: '#/components/schemas/ExternalOrder'
+        match_result:
+          $ref: '#/components/schemas/ApiExternalMatchResult'
+        fees:
+          $ref: '#/components/schemas/FeeTake'
+        send:
+          $ref: '#/components/schemas/ApiExternalAssetTransfer'
+        receive:
+          $ref: '#/components/schemas/ApiExternalAssetTransfer'
+        price:
+          $ref: '#/components/schemas/ApiTimestampedPrice'
+        timestamp:
+          type: integer
+          description: Unix timestamp in milliseconds since epoch
+
+    ApiSignedExternalQuote:
+      type: object
+      properties:
+        quote:
+          $ref: '#/components/schemas/ApiExternalQuote'
+        signature:
+          type: string
+          description: Signature of the quote, a base64-encoded HMAC-SHA256 hash of the quote 
+          by the Relayer's admin API key
```

[Swagger](https://editor.swagger.io/) parser errors here due to the line break:
```suggestion
          description: Signature of the quote, a base64-encoded HMAC-SHA256 hash of the quote by the Relayer's admin API key
```